lightningd/jsonrpc.c

```diff
@@ -231,11 +234,24 @@ static void json_help(struct command *cmd,
 		return;
 	}
 
+	cmd->mode = CMD_USAGE;
+	cmd->ok = &ok;
 	json_array_start(response, "help");
 	for (i = 0; i < num_cmdlist; i++) {
+		char *usage = tal_strdup(cmd, cmdlist[i]->name);
+		/* We have to skip all the ones that don't call param() in
+		 * their handler. */
+		if (strcmp(cmdlist[i]->name, "help")) {
+			ok = true;
+			cmdlist[i]->dispatch(cmd, NULL, NULL);
+			if (!ok)
+				return;
```

Comment:
This looks really weird.  I'm not sure how we ever set cmd->ok to false.  Actually, seems we don't?  Maybe remove the field altogether...

Reply:
Ah, I see it later.  Still not sure when help() would fail in a command though?

Reply:
You're right. I over thought this.